pontoon/sync/formats/l20n.py

```diff
@@ -0,0 +1,183 @@
+from __future__ import absolute_import
+
+import codecs
+import copy
+import logging
+import os
+
+from l20n.format.parser import L20nParser
+from l20n.format.serializer import Serializer
+from l20n.format import ast
+
+from pontoon.sync import SyncError
+from pontoon.sync.formats.base import ParsedResource
+from pontoon.sync.vcs.models import VCSTranslation
+
+log = logging.getLogger(__name__)
+
+
+class L20NEntity(VCSTranslation):
+    """
+    Represents entities in l20n (without its attributes).
+    """
+    def __init__(self, key, source_string, source_string_plural, strings, comments=None, order=None):
+        super(L20NEntity, self).__init__(
+            key=key,
+            source_string=source_string,
+            source_string_plural=source_string_plural,
+            strings=strings,
+            comments=comments or [],
+            fuzzy=False,
+            order=order,
+        )
+
+    def __repr__(self):
+        return '<L20NEntity {key}>'.format(key=self.key.encode('utf-8'))
+
+
+class L20NResource(ParsedResource):
+    def __init__(self, path, locale, source_resource=None):
+        self.path = path
+        self.locale = locale
+        self.entities = {}
+        self.source_resource = source_resource
+
+        # Copy entities from the source_resource if it's available.
+        if source_resource:
+            for key, entity in source_resource.entities.items():
+                self.entities[key] = L20NEntity(
+                    entity.key,
+                    '',
+                    '',
+                    {},
+                    copy.copy(entity.comments),
+                    entity.order
+                )
+
+        with codecs.open(path, 'r', 'utf-8') as resource:
+            self.structure = L20nParser().parse(resource.read())
+            order = 0
+            comments = []
+            for obj in self.structure.body:
+                if obj.type == 'Entity':
+                    if obj.value.type == 'String':
```

Comment:
Use `if obj.value and obj.value.type == 'String':` or you might get `AttributeError: 'NoneType' object has no attribute 'type'`.